framework/Models/PostProcessors/validationAlgorithms/Representativity.py

```diff
@@ -0,0 +1,247 @@
+      results to specified dataObject
+      @ In, inputIn, list, dictionary of data to process
+      @ Out, outputDict, dict, dictionary containing the post-processed results
+    """
+    dataSets = [data for _, _, data in inputIn['Data']]
+    pivotParameter = self.pivotParameter
+    names=[]
+    if isinstance(inputIn['Data'][0][-1], xr.Dataset):
+      names = [inp[-1].attrs['name'] for inp in inputIn['Data']]
+      if len(inputIn['Data'][0][-1].indexes) and self.pivotParameter is None:
+        if 'dynamic' not in self.dynamicType: #self.model.dataType:
+          self.raiseAnError(IOError, "The validation algorithm '{}' is not a dynamic model but time-dependent data has been inputted in object {}".format(self._type, inputIn['Data'][0][-1].name))
+        else:
+          pivotParameter = self.pivotParameter
+    evaluation ={k: np.atleast_1d(val) for k, val in  self._evaluate(dataSets, **{'dataobjectNames': names}).items()}
+    if pivotParameter:
+      if len(dataSets[0][pivotParameter]) != len(list(evaluation.values())[0]):
+        self.raiseAnError(RuntimeError, "The pivotParameter value '{}' has size '{}' and validation output has size '{}'".format( len(dataSets[0][self.pivotParameter]), len(evaluation.values()[0])))
+      if pivotParameter not in evaluation:
+        evaluation[pivotParameter] = dataSets[0][pivotParameter]
+    return evaluation
+
+  def _evaluate(self, datasets, **kwargs):
+    """
+      Main method to "do what you do".
+      @ In, datasets, list, list of datasets (data1,data2,etc.) to used.
+      @ In, kwargs, dict, keyword arguments
+      @ Out, outputDict, dict, dictionary containing the results {"feat"_"target"_"metric_name":value}
+    """
+    # self.stat.run({'targets':{self.target:xr.DataArray(self.functionS.evaluate(tempDict)[self.target])}})[self.computationPrefix +"_"+self.target]
+    for data in datasets:
+      sen = self.stat.run(data)
+    names = kwargs.get('dataobjectNames')
+    outs = {}
+    for feat, targ, param, targParam in zip(self.features, self.targets, self.Parameters, self.targetParameters):
+      featData = self._getDataFromDatasets(datasets, feat, names)
+      targData = self._getDataFromDatasets(datasets, targ, names)
+      Parameters = self._getDataFromDatasets(datasets, param, names)
+      targetParameters = self._getDataFromDatasets(datasets, targParam, names)
+      # senFOMs = partialDerivative(featData.data,np.atleast_2d(Parameters.data)[0,:],'x1')
+      senFOMs = np.atleast_2d(Parameters[0])#.data
+      senMeasurables = np.atleast_2d(targetParameters[0])
+      covParameters = senFOMs @ senMeasurables.T
+      for metric in self.metrics:
+        name = "{}_{}_{}".format(feat.split("|")[-1], targ.split("|")[-1], metric.name)
+        outs[name] = metric.evaluate((featData, targData), senFOMs = senFOMs, senMeasurables=senMeasurables, covParameters=covParameters)
+    return outs
+
+  def _getDataFromDatasets(self, datasets, var, names=None):
```

Comment:
I guess you can remove this function, and you can call it in the Validation base class